### PR TITLE
Migrated profile_phone_number to regions from resty to request helpers

### PR DESCRIPTION
**Pull request by ezilber-akamai**
## 📝 Description

Migrated profile_phone_number to regions to request helpers

### CODE REVIEW

**Review by yec-akamai — APPROVED** [review]
LGTM

**Review by ykim-1 — APPROVED** [review]
LGTM, nice work!

File: pagination.go [hunk 1]
```diff
@@ -91,41 +90,6 @@ type PagedResponse interface {
 	castResult(*resty.Request, string) (int, int, error)
 }
 
-// listHelper abstracts fetching and pagination for GET endpoints that
-// do not require any Ids (top level endpoints).
-// When opts (or opts.Page) is nil, all pages will be fetched and
-// returned in a single (endpoint-specific)PagedResponse
-// opts.results and opts.pages will be updated from the API response
-func (c *Client) listHelper(ctx context.Context, pager PagedResponse, opts *ListOptions, ids ...any) error {
```
Comment (lgarber-akamai) [inline pagination.go:99]: 🎉

**Review by lgarber-akamai — APPROVED** [review]
Validated that the fixtures are free of sensitive data and all PagedResponse structs are preserved. Nice work!

**Pull request merged**